From 016d6b357484ea34337f0c80ce74e77601cf3d1a Mon Sep 17 00:00:00 2001
From: Bastian Koppelmann <kbastian@mail.uni-paderborn.de>
Date: Tue, 30 Aug 2022 12:50:05 +0200
Subject: [PATCH] Fix:graphics:svg_debug: Fix -Wincompatible-pointer-types
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

we are assigning a char* to int*. So lets explicitly cast it to int*.
This fixes:

graphics_svg_debug.c:173:16: warning: assignment to ‘unsigned int *’ from incompatible pointer type ‘unsigned char *’ [-Wincompatible-pointer-types]
  173 |     gc->dashed = dash_list;
      |                ^
---
 navit/graphics/svg_debug/graphics_svg_debug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/navit/graphics/svg_debug/graphics_svg_debug.c b/navit/graphics/svg_debug/graphics_svg_debug.c
index 348906ad8..23ce4a297 100644
--- a/navit/graphics/svg_debug/graphics_svg_debug.c
+++ b/navit/graphics/svg_debug/graphics_svg_debug.c
@@ -170,7 +170,7 @@ static void svg_debug_gc_set_linewidth(struct graphics_gc_priv *gc, int w) {
 
 static void svg_debug_gc_set_dashes(struct graphics_gc_priv *gc, int w,
                                     int offset, unsigned char *dash_list, int n) {
-    gc->dashed = dash_list;
+    gc->dashed = (unsigned int*)dash_list;
     gc->is_dashed = TRUE;
     if (gc->graphics_gc_methods_proxy->gc_set_dashes) {
         gc->graphics_gc_methods_proxy->gc_set_dashes(gc->graphics_gc_priv_proxy,
-- 
2.50.0

